coderd/parameters.go

```diff
@@ -297,10 +335,31 @@ func (api *API) handleStaticParameters(rw http.ResponseWriter, r *http.Request,
 					Detail:   "To restore full functionality, please re-import the terraform as a new template version.",
 				},
 			}
-	})
+	}
+	if listen {
+		api.handleParameterWebsocket(rw, r, initial, staticRender)
+	} else {
+		api.handleParameterEvaluate(rw, r, initial, staticRender)
+	}
+}
+
+func (api *API) handleParameterEvaluate(rw http.ResponseWriter, r *http.Request, initial codersdk.DynamicParametersRequest, render previewFunction) {
+	ctx := r.Context()
+
+	// Send an initial form state, computed without any user input.
+	result, diagnostics := render(ctx, initial.OwnerID, initial.Inputs)
+	response := codersdk.DynamicParametersResponse{
+		ID:          -1, // Always start with -1.
```

Comment:
nit: while technically a breaking change, we could add `omitzero` as this is an experimental feature.